pinot-core/src/main/java/org/apache/pinot/core/data/manager/offline/DimensionTableDataManager.java

```diff
@@ -117,6 +134,27 @@ public void removeSegment(String segmentName) {
     }
   }
 
+  @Override
+  protected void doShutdown() {
+    closeDimensionTable(_dimensionTable, true);
+  }
+
+  private void closeDimensionTable(DimensionTable dimensionTable, boolean releaseSegments) {
+    if (dimensionTable != null) {
+      try {
+        dimensionTable.close();
+        if (releaseSegments) {
+          List<SegmentDataManager> segmentDataManagers = acquireAllSegments();
+          for (SegmentDataManager segmentDataManager: segmentDataManagers) {
+            releaseSegment(segmentDataManager);
+          }
```

Comment:
This is incorrect. Acquiring then release is no-op, and the segments won't be released in the end.
Instead, when preload is disabled, we can track the `SegmentDataManagers` and release them here.